3634/3/app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,32 @@
+require 'pry'
+
+class CommentsController < ApplicationController
+  before '/restaurants/:id/*' do
+    if !logged_in?
+      flash[:message] = "You're not permitted to leave comment. Please, login"
+      redirect '/'
+    elsif restaurant_not_exist?
```

Comment:
you'll already have the handler for the error `ActiveRecord::RecordNotFound`
remove the checking and just create a private controller method like `verify_restaurant!` and move restaurant verification logic in the method. and just call it in the controller